constraints_gpu.txt

```diff
@@ -19,139 +19,139 @@ dm-tree==0.1.8
-grpc-google-iam-v1==0.13.0
-grpcio==1.62.1
+google-resumable-media==2.7.1
+googleapis-common-protos==1.63.2
+grain-nightly==0.0.7
+grpc-google-iam-v1==0.13.1
+grpcio==1.64.1
 grpcio-status==1.48.2
 gviz-api==1.10.0
-h5py==3.10.0
-idna==3.6
+h5py==3.11.0
+idna==3.7
 immutabledict==4.2.0
 importlab==0.8.1
-importlib_resources==6.3.0
+importlib_resources==6.4.0
 iniconfig==2.0.0
 isort==5.13.2
-jax==0.4.25
-jaxlib==0.4.25
-jaxtyping==0.2.28
-Jinja2==3.1.3
-keras==2.13.1
-libclang==16.0.6
-libcst==1.2.0
-Markdown==3.5.2
+jax==0.4.30
+jaxlib==0.4.30
+jaxtyping==0.2.30
+Jinja2==3.1.4
+keras==3.3.3
+libclang==18.1.1
+libcst==1.4.0
+Markdown==3.6
 markdown-it-py==3.0.0
 MarkupSafe==2.1.5
 mccabe==0.7.0
 mdurl==0.1.2
 ml-collections==0.1.1
 ml-dtypes==0.3.2
-ml_goodput_measurement==0.0.2
+ml_goodput_measurement==0.0.3
 mlperf-logging==3.0.0
-more-itertools==10.2.0
+more-itertools==10.3.0
 msgpack==1.0.8
 msgspec==0.18.6
 mypy-extensions==1.0.0
 nest-asyncio==1.6.0
 networkx==3.1
 ninja==1.11.1.1
-numpy==1.24.3
-nvidia-cublas-cu12==12.4.2.65
-nvidia-cuda-cupti-cu12==12.4.99
-nvidia-cuda-nvcc-cu12==12.4.99
-nvidia-cuda-nvrtc-cu12==12.4.99
-nvidia-cuda-runtime-cu12==12.4.99
-nvidia-cudnn-cu12==8.9.7.29
-nvidia-cufft-cu12==11.2.0.44
-nvidia-cusolver-cu12==11.6.0.99
-nvidia-cusparse-cu12==12.3.0.142
-nvidia-nccl-cu12==2.19.3
-nvidia-nvjitlink-cu12==12.4.99
+numpy==1.26.4
+nvidia-cublas-cu12==12.5.2.13
+nvidia-cuda-cupti-cu12==12.5.39
+nvidia-cuda-nvcc-cu12==12.5.40
+nvidia-cuda-nvrtc-cu12==12.5.40
+nvidia-cuda-runtime-cu12==12.5.39
+nvidia-cudnn-cu12==9.1.1.17
+nvidia-cufft-cu12==11.2.3.18
+nvidia-cusolver-cu12==11.6.2.40
+nvidia-cusparse-cu12==12.4.1.24
+nvidia-nccl-cu12==2.22.3
+nvidia-nvjitlink-cu12==12.5.40
 oauthlib==3.2.2
 opt-einsum==3.3.0
-optax==0.2.1
-orbax-checkpoint==0.5.11 
+optax==0.2.2
+orbax-checkpoint==0.5.20
 packaging==24.0
-pandas==2.2.1
-platformdirs==4.2.0
-pluggy==1.4.0
+pandas==2.2.2
+platformdirs==4.2.2
+pluggy==1.5.0
 portpicker==1.6.0
-pre-commit==3.7.0
+pre-commit==3.7.1
 promise==2.3
-proto-plus==1.23.0
+proto-plus==1.24.0
 protobuf==3.20.3
-psutil==5.9.8
-pyasn1==0.5.1
-pyasn1-modules==0.3.0
-pycnite==2023.10.11
-pydantic==1.10.14
+psutil==6.0.0
+pyasn1==0.6.0
+pyasn1_modules==0.4.0
+pycnite==2024.6.13
+pydantic==2.7.4
 pydot==2.0.0
 pyglove==0.4.4
-Pygments==2.17.2
+Pygments==2.18.0
 pyink==24.3.0
-pylint==3.1.0
+pylint==3.2.3
 pyparsing==3.1.2
-pytest==8.1.1
+pytest==8.2.2
 python-dateutil==2.9.0.post0
-pytype==2024.3.11
+pytype==2024.4.11
 pytz==2024.1
-PyYAML==6.0.1
-requests==2.31.0
-requests-oauthlib==1.4.0
+PyYAML==5.4.1
+requests==2.32.3
+requests-oauthlib==2.0.0
 rich==13.7.1
 rsa==4.9
-scipy==1.12.0
+scipy==1.13.1
 sentencepiece==0.1.97
 seqio==0.0.19
-shapely==2.0.3
+shapely==2.0.4
 six==1.16.0
 tabulate==0.9.0
-tensorboard==2.13.0
+tensorboard==2.16.2
 tensorboard-data-server==0.7.2
 tensorboard_plugin_profile==2.15.1
 tensorboardX==2.6.2.2
-tensorflow==2.13.1
-tensorflow-datasets==4.9.4
+tensorflow==2.16.1
```

Comment:
tensorflow version has been deliberately held back due to https://github.com/google/maxtext/issues/516

could you check if the issue has been resolved now? TL;DR OOMs on A3+ machines when using C4 dataset with tf-data

Reply:
Withheld some package versions and there seems to be no issues. Waiting for the final check to pass before merging this PR.